gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingExtractor.java

```diff
@@ -0,0 +1,216 @@
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.Watermark;
+import gobblin.source.extractor.WatermarkSerializerHelper;
+import gobblin.source.extractor.extract.EventBasedExtractor;
+import gobblin.source.extractor.StreamingExtractor;
+import gobblin.source.extractor.RecordEnvelope;
+import gobblin.source.extractor.CheckpointableWatermark;
+import gobblin.source.extractor.ComparableWatermark;
+import gobblin.source.extractor.extract.LongWatermark;
+import gobblin.metrics.Tag;
+import gobblin.source.extractor.extract.kafka.SimpleKafkaSchemaRegistry;
+import gobblin.writer.WatermarkStorage;
+
+
+/**
+ * An implementation of {@link StreamingExtractor} from which reads from Kafka and returns records as an array of bytes.
+ *
+ * @author Shrikanth Shankar
+ *
+ *
+ */
+public class KafkaSimpleStreamingExtractor extends EventBasedExtractor<String, RecordEnvelope<byte[]>> implements StreamingExtractor<String, byte[]>, WatermarkStorage {
+
+  public static class KafkaWatermark implements CheckpointableWatermark {
+    TopicPartition _topicPartition;
+    LongWatermark _lwm;
+
+    @VisibleForTesting
+    public KafkaWatermark(TopicPartition topicPartition, LongWatermark lwm) {
+      _topicPartition = topicPartition;
+      _lwm = lwm;
+    }
+
+    @Override
+    public String getSource() {
+      return _topicPartition.toString();
+    }
+
+    @Override
+    public ComparableWatermark getWatermark() {
+      return _lwm;
+    }
+
+    @Override
+    public short calculatePercentCompletion(Watermark lowWatermark, Watermark highWatermark) {
+      return 0;
+    }
+
+    @Override
+    public JsonElement toJson() {
+      return WatermarkSerializerHelper.convertWatermarkToJson(this);
+    }
+
+    @Override
+    public int compareTo(CheckpointableWatermark o) {
+      Preconditions.checkArgument(o instanceof KafkaWatermark);
+      KafkaWatermark ko = (KafkaWatermark)o;
+      Preconditions.checkArgument(_topicPartition.equals(ko._topicPartition));
+      return _lwm.compareTo(ko._lwm);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == null)
+        return false;
+      if (!(obj instanceof KafkaWatermark))
+          return false;
+      return this.compareTo((CheckpointableWatermark)obj) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      return _topicPartition.hashCode() * prime + _lwm.hashCode();
+    }
+
+    public TopicPartition getTopicPartition() {return _topicPartition;};
+
+    public LongWatermark getLwm() {return  _lwm;};
+  }
+
+
+  private final KafkaSchemaRegistry<String, String> _kafkaSchemaRegistry;
+  private Consumer<String, byte[]> _consumer;
+  private TopicPartition _partition;
+  private Iterator<ConsumerRecord<String, byte[]>> _records;
+  long _rowCount = 0;
+
+  public KafkaSimpleStreamingExtractor(WorkUnitState state) {
+    super(state);
+    _kafkaSchemaRegistry = new SimpleKafkaSchemaRegistry(state.getProperties());
```

Comment:
I assume this should be changeable. Typically, we add a config property with the factory interface. You can have a default value of "SimpleKafkaSchemaRegistry". We also have a pattern for being able to add human-readable names instead of class names for better usability. You can have a look at  gobblin.runtime.JobContext.createStateStore(Properties) for an example.